### PR TITLE
feat: added user metadata to eventsources

### PR DESCRIPTION
**Pull request by VaibhavPage**

### CODE REVIEW

**Comment by VaibhavPage** [comment]
#795

File: pkg/apis/eventsource/v1alpha1/types.go [hunk 1]
```diff
@@ -183,7 +183,11 @@ type CalendarEventSource struct {
 	Timezone string `json:"timezone,omitempty" protobuf:"bytes,4,opt,name=timezone"`
 	// UserPayload will be sent to sensor as extra data once the event is triggered
 	// +optional
+	// Deprecated. Please use Metadata instead.
 	UserPayload json.RawMessage `json:"userPayload,omitempty" protobuf:"bytes,5,opt,name=userPayload,casttype=encoding/json.RawMessage"`
+	// Metadata holds the user defined metadata which will passed along the event payload.
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty" protobuf:"bytes,6,opt,name=metadata"`
```
Comment (whynowy) [inline pkg/apis/eventsource/v1alpha1/types.go:190]: `bytes,6,rep,name=metadata`

File: pkg/apis/eventsource/v1alpha1/types.go [hunk 2]
```diff
@@ -195,6 +199,9 @@ type FileEventSource struct {
 	WatchPathConfig WatchPathConfig `json:"watchPathConfig" protobuf:"bytes,2,opt,name=watchPathConfig"`
 	// Use polling instead of inotify
 	Polling bool `json:"polling,omitempty" protobuf:"varint,3,opt,name=polling"`
+	// Metadata holds the user defined metadata which will passed along the event payload.
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty" protobuf:"bytes,4,opt,name=metadata"`
```
Comment (whynowy) [inline pkg/apis/eventsource/v1alpha1/types.go:204]: rep

File: pkg/apis/eventsource/v1alpha1/types.go [hunk 3]
```diff
@@ -220,6 +227,9 @@ type ResourceEventSource struct {
 	// EventTypes is the list of event type to watch.
 	// Possible values are - ADD, UPDATE and DELETE.
 	EventTypes []ResourceEventType `json:"eventTypes" protobuf:"bytes,4,rep,name=eventTypes,casttype=ResourceEventType"`
+	// Metadata holds the user defined metadata which will passed along the event payload.
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty" protobuf:"bytes,5,opt,name=metadata"`
```
Comment (whynowy) [inline pkg/apis/eventsource/v1alpha1/types.go:232]: rep

**Pull request merged**